script.js

```diff
@@ -1,4 +1,336 @@
+    var randomIndex = getRandomIndex(deckToShuffle.length);
+    var randomCard = deckToShuffle[randomIndex];
+    var currentCard = deckToShuffle[currentIndex];
+    deckToShuffle[currentIndex] = randomCard;
+    deckToShuffle[randomIndex] = currentCard;
+    currentIndex = currentIndex + 1;
+  }
+  return deckToShuffle;
+};
+
+var shuffledDeck = shuffleCards(makeDeck());
+
+// Function: deal 1 card to player, 1 card to computer. Repeat one more time.
+var dealStartingCards = function () {
+  var dealIndex = 0;
+  //start loop to deal cards twice to each player
+  while (dealIndex < 2) {
+    //deal 1 card to user
+    userHand.push(shuffledDeck.pop());
+    //update global variable for user's total points
+    updateUserTotal();
+    //deal 1 card to computer
+    computerHand.push(shuffledDeck.pop());
+    //update global variable for computer's total points
+    updateComputerTotal();
+    dealIndex += 1;
+  }
+  console.log("computerHand is:");
+  console.log(computerHand);
+  console.log("userHand is:");
+  console.log(userHand);
+};
+
+// Function: check if user's starting hand is Blackjack
+var isBlackjack = function () {
+  //evaluate to TRUE statement which can be used in main function as a conditional
+  return userHand[0].value + userHand[1].value == 21;
+};
+
+// function to output message for a player's cards
+var outputCards = function (aHand) {
+  //set a loop to run through a player's entire hand and output all the cards' names and suits
+  var cardsIndex = 0;
+  var outputCardsMessage = "";
+  while (cardsIndex < aHand.length) {
+    outputCardsMessage =
+      outputCardsMessage +
+      aHand[cardsIndex].name +
+      " of " +
+      aHand[cardsIndex].suit +
+      "<br>";
+    cardsIndex += 1;
+  }
+  console.log("outputUserCards:");
+  console.log(outputCardsMessage);
+  return outputCardsMessage;
+};
+
+// function calculate total value of cards in a hand, also to convert Ace value from 11 to 1 when total is more than 21
+var calcHandTotal = function (hand) {
+  //set a loop to go through the hand's cards and add the value of the cards to get total points
+  var handTotal = 0;
+  for (var i = 0; i < hand.length; i += 1) {
+    handTotal = handTotal + hand[i].value;
+  }
+  //execute function to check for aces in hand to find out number of aces in hand to be subtracted if total points more than 21
+  var numOfAces = checkForAce(hand);
+  //if hand total is more than 21 and there is 1 or more aces in hand, then convert 1 ace at a time from 11 to 1 points and recount hand total and repeat
+  while (handTotal > 21 && numOfAces > 0) {
+    handTotal = handTotal - 10;
+    numOfAces = numOfAces - 1;
+  }
+  return handTotal;
+};
+
+// function update user total
+var updateUserTotal = function () {
+  userTotal += userHand[userHand.length - 1].value;
+};
+
+// function update computer total
+var updateComputerTotal = function () {
+  computerTotal += computerHand[computerHand.length - 1].value;
+};
```

Comment:
both of this have the same logic , we could reuse it by having different parameter input